heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -159,84 +197,267 @@ export default class JestPlugin implements IHeftTaskPlugin<IJestPluginOptions> {
-        options?.extensionForTests || typeScriptConfigurationJson?.emitCjsExtensionForCommonJS
-          ? '.cjs'
-          : '.js',
-      isTypeScriptProject: !!partialTsconfigFile
-    });
-    terminal.writeVerboseLine(`Wrote ${HEFT_JEST_DATA_FILENAME} file`);
+    const {
+      // Config.Argv is weakly typed.  After updating the jestArgv object, it's a good idea to inspect "globalConfig"
+      // in the debugger to validate that your changes are being applied as expected.
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      globalConfig,
+      results: jestResults
+    } = await runCLI(jestArgv, [buildFolderPath]);
+
+    if (jestResults.numFailedTests > 0) {
+      logger.emitError(
+        new Error(
+          `${jestResults.numFailedTests} Jest test${jestResults.numFailedTests > 1 ? 's' : ''} failed`
+        )
+      );
+    } else if (jestResults.numFailedTestSuites > 0) {
+      logger.emitError(
+        new Error(
+          `${jestResults.numFailedTestSuites} Jest test suite${
+            jestResults.numFailedTestSuites > 1 ? 's' : ''
+          } failed`
+        )
+      );
+    }
   }
 
   /**
    * Runs Jest using the provided options.
    */
-  private async _runJestAsync(
+  private async _runJestWatchAsync(
     taskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
-    options: IJestPluginOptions
+    options: IJestPluginOptions,
+    requestRun: () => void
   ): Promise<void> {
     const logger: IScopedLogger = taskSession.logger;
     const terminal: ITerminal = logger.terminal;
-    terminal.writeLine(`Using Jest version ${getVersion()}`);
 
-    // Write the jest data file used by the BuildTransformer
-    await this._setupJestAsync(taskSession, heftConfiguration, options);
+    const pendingTestRuns: Set<IPendingTestRun> = this._pendingTestRuns;
+    this._requestRun = requestRun;
+
+    if (!this._jestPromise) {
+      // Monkey-patch Jest's watch mode so that we can orchestrate it.
+      const jestCoreDir: string = path.dirname(require.resolve('@jest/core'));
+
+      const wrappedStdOut: TerminalWritableStream = new TerminalWritableStream(terminal);
+      this._jestOutputStream = wrappedStdOut;
+
+      // Shim watch so that we can intercept the output stream
+      const watchModulePath: string = path.resolve(jestCoreDir, 'watch.js');
+      const watchModule: {
+        default: IJestWatch;
+      } = require(watchModulePath);
+      const { default: originalWatch } = watchModule;
+
+      const watch: IJestWatch = function patchedWatch(
+        this: void,
+        initialGlobalConfig: { seed: JestPlugin },
+        contexts: unknown,
+        outputStream: NodeJS.WriteStream,
+        hasteMapInstances: EventEmitter[],
+        stdin: NodeJS.ReadStream | undefined,
+        hooks: unknown,
+        filter: unknown
+      ): Promise<void> {
+        // The `seed` property is one of the few values that gets passed as-is from the argv into
```

Comment:
As amusing as this approach is, I should probably just patch `jest-config`'s `readConfigs` method to forward some `argv[HeftJestPluginSymbol]` onto `globalConfig`.